bentoml/utils/pip_pkg.py

```diff
@@ -0,0 +1,202 @@
+        self.is_local = is_local
+        self.is_pkg = is_pkg
+
+
+class ModuleManager(object):
+    def __init__(self):
+        super(ModuleManager, self).__init__()
+        self.pip_pkg_map = {}
+        self.pip_module_map = {}
+        self.setuptools_module_set = set()
+        for dist in pkg_resources.working_set:
+            self.pip_pkg_map[dist._key] = dist._version
+            for mn in dist._get_metadata("top_level.txt"):
+                if dist._key != "setuptools":
+                    self.pip_module_map.setdefault(mn, []).append(
+                        (dist._key, dist._version)
+                    )
+                else:
+                    self.setuptools_module_set.add(mn)
+
+        self.nonlocal_package_path = set()
+
+        self.searched_modules = {}
+        for m in pkgutil.iter_modules():
+            if m[1] not in self.searched_modules:
+                path = m[0].path
+                is_local = self.is_local_path(path)
+                self.searched_modules[m[1]] = ModuleInfo(m[1], path, is_local, m[2])
+
+    def verify_pkg(self, pkg_name, pkg_version):
+        if pkg_name not in self.pip_pkg_map:
+            # package does not exist in the current python session
+            return EPP_PKG_NOT_EXIST
+        if pkg_version and pkg_version != self.pip_pkg_map[pkg_name]:
+            # package version is different from the version being used
+            # in the current python session
+            return EPP_PKG_VERSION_MISMATCH
+        return EPP_NO_ERROR
+
+    def seek_pip_dependencies(self, target_py_file_path):
+        work = DepSeekWork(self, target_py_file_path)
+        work.do()
+        requirements = {}
+        for _, pkg_info_list in work.dependencies.items():
+            for pkg_name, pkg_version in pkg_info_list:
+                requirements[pkg_name] = pkg_version
+
+        return requirements, work.unknown_module_set
+
+    def is_local_path(self, path):
+        if path in self.nonlocal_package_path:
+            return False
+
+        dir_name = os.path.split(path)[1]
+
+        if (
+            "site-packages" in path
+            or "anaconda" in path
+            or path.endswith("packages")
+            or dir_name == "bin"
+            or dir_name.startswith("lib")
+            or dir_name.startswith("python")
+            or dir_name.startswith("plat")
+        ):
+            self.nonlocal_package_path.add(path)
+            return False
+
+        return True
+
+
+class DepSeekWork(object):
+    def __init__(self, module_manager, target_py_file_path):
+        super(DepSeekWork, self).__init__()
+        self.module_manager = module_manager
+        self.target_py_file_path = target_py_file_path
+
+        self.dependencies = {}
+        self.unknown_module_set = set()
+        self.parsed_module_set = set()
+
+    def do(self):
+        self.seek_in_file(self.target_py_file_path)
+
+    def seek_in_file(self, file_path):
+        if not os.path.isfile(file_path):
```

Comment:
why do we need this check here?  and should it show a warning when it happens?

Reply:
No need to check, remove it.